### PR TITLE
chore(releasing): Prepare v0.43.0

### PR DESCRIPTION
**Pull request by pront**
## Change Type
- [ ] Bug fix
- [ ] New feature
- [x] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?
- [ ] Yes
- [x] No

## Does this PR include user facing changes?

- [ ] Yes. Please add a changelog fragment based on our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [x] No. A maintainer will apply the "no-changelog" label to this PR.

## Checklist
- [ ] Please read our [Vector contributor resources](https://github.com/vectordotdev/vector/tree/master/docs#getting-started).
- [x] If this PR introduces changes Vector dependencies (modifies `Cargo.lock`), please
  run `dd-rust-license-tool write` to regenerate the [license inventory](https://github.com/vectordotdev/vrl/blob/main/LICENSE-3rdparty.csv) and commit the changes (if any). More details [here](https://crates.io/crates/dd-rust-license-tool).

## References

https://github.com/vectordotdev/vector/issues/21893

### CODE REVIEW

File: website/cue/reference/releases/0.43.0.cue [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+
+- Fixes the `to_float` function to return an error instead of `f64::INFINITY` when parsing [non-normal](https://doc.rust-lang.org/std/primitive.f64.html#method.is_normal) numbers. (https://github.com/vectordotdev/vrl/pull/1107)
+
+### New Features
+
+- The `decrypt` and `encrypt` VRL functions now support aes-siv (RFC 5297) encryption and decryption. (https://github.com/vectordotdev/vrl/pull/1100)
+
+### Enhancements
+
+- `decode_punycode` and `encode_punycode` with `validate` flag set to false should be faster now, in cases when input data needs no encoding or decoding. (https://github.com/vectordotdev/vrl/pull/1104)
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:32]:
```suggestion
- `decode_punycode` and `encode_punycode` with the `validate` flag set to false should be faster now, in cases when input data needs no encoding or decoding. (https://github.com/vectordotdev/vrl/pull/1104)
```

File: website/cue/reference/releases/0.43.0.cue [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+- `vrl::value::Value` now implements `PartialCmp` that first checks whether the enum discriminants
+	(that both are floats for example), and if they are calls `partial_cmp` on the inner values.
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:34]: Can you explain what this sentence means? Especially "checks whether the enum discriminants (that both are floats for example)"

File: website/cue/reference/releases/0.43.0.cue [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+	Otherwise, it will return `None`. (https://github.com/vectordotdev/vrl/pull/1117)
+- The `encode_proto` function was enhanced to automatically convert valid string fields to numeric proto
+	fields. (https://github.com/vectordotdev/vrl/pull/1114)
+
+### Fixes
+
+- The `parse_groks` VRL function and Datadog grok parsing now catch the panic coming from `rust-onig` on too many regex match retries, and handles it as a custom error. (https://github.com/vectordotdev/vrl/pull/1079)
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:41]:
```suggestion
- The `parse_groks` VRL function and Datadog grok parsing now catch the panic coming from `rust-onig` on too many regex match retries and handle it as a custom error. (https://github.com/vectordotdev/vrl/pull/1079)
```

File: website/cue/reference/releases/0.43.0.cue [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+- `encode_punycode` with `validate` flag set to false should be more consistent with `validate` set to true, turning all uppercase character to lowercase besides doing punycode encoding (https://github.com/vectordotdev/vrl/pull/1115)
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:42]:
```suggestion
- `encode_punycode` with the `validate` flag set to false should be more consistent with when `validate` is set to true, turning all uppercase character to lowercase as well as doing punycode encoding (https://github.com/vectordotdev/vrl/pull/1115)
```

File: website/cue/reference/releases/0.43.0.cue [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+				Allows for chunked gelf decoding in message-based sources, such as UDP sockets or unix datagram sockets. Implementation is based on [Graylog's documentation](https://go2docs.graylog.org/5-0/getting_in_log_data/gelf.html#GELFviaUDP).
+
+				This framing method can be configured via the `framing.method = "chunked_gelf"` option in the source configuration.
+				"""
+			contributors: ["jorgehermo9"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				The `sample` transform can now take in a `group_by` configuration option that will allow logs with unique values for the patterns passed in to be sampled independently. This can reduce the complexity of the topology, since users would no longer need to create separate samplers with similar configuration to handle different log streams.
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:87]:
```suggestion
				The `sample` transform can now take in a `group_by` configuration option that allows logs with unique values for the patterns passed in to be sampled independently. This can reduce the complexity of the topology, since users no longer need to create separate samplers with similar configurations to handle different log streams.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+				- `max_retries`: Maximum number of connection retries.
+				- `connection_timeout_secs`: Time limit to establish a connection.
+				- `keep_alive_secs`: Keep-alive interval for each broker connection.
+				"""
+			contributors: ["FRosner"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				The `http sink` will now retry requests when the response is a request timeout (HTTP 408).
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:114]:
```suggestion
				The `http sink` now retries requests when the response is a request timeout (HTTP 408).
```

File: website/cue/reference/releases/0.43.0.cue [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+			type: "feat"
+			description: """
+				The `opentelemetry` source can now be configured to enrich log events with HTTP headers received in the OTLP/HTTP request.
+				"""
+			contributors: ["jblazquez"]
+		},
+		{
+			type: "feat"
+			description: """
+				Introduce a new exclusive_route transform, which functions as a switch statement to route events based on user-defined conditions.
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:142]:
```suggestion
				Introduce a new `exclusive_route` transform, which functions as a switch statement to route events based on user-defined conditions.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+				"""
+			contributors: ["pront"]
+		},
+		{
+			type: "fix"
+			description: """
+				Vector now supports YAML merges in configuration files, a part of the YAML 1.1
+				specification. This functionality is useful for reducing the size of transform
+				configurations. See YAML documentation [here](https://yaml.org/type/merge.html).
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:151]:
```suggestion
				configurations. See the [YAML documentation](https://yaml.org/type/merge.html).
```

File: website/cue/reference/releases/0.43.0.cue [hunk 9]
```diff
@@ -0,0 +1,449 @@
...
+			description: """
+				EDE code 30 (Invalid Query Type) (added in https://datatracker.ietf.org/doc/draft-ietf-dnsop-compact-denial-of-existence/04/) has been added to recognized EDE codes and has correct `purpose` attached to it.
+				"""
+			contributors: ["esensar"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				Add VECTOR_HOSTNAME env variable to override the hostname used in the Vector events and internal metrics.
+				This is useful when Vector is running on a system where the hostname is not meaningful, e.g., in a container (Kubernetes).
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:173]:
```suggestion
				This is useful when Vector is running on a system where the hostname is not meaningful, such as in a container (Kubernetes).
```

File: website/cue/reference/releases/0.43.0.cue [hunk 10]
```diff
@@ -0,0 +1,449 @@
...
+			description: """
+				Vector now supports two additional back-ends for loading secrets: `file`, for reading a set of
+				secrets from a JSON file, and `directory`, for loading secrets from a list of files.
+				"""
+			contributors: ["tie"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				Add Gzip compression support to the`gcp_chronicle_unstructured` sink. See official documentation [here](https://cloud.google.com/chronicle/docs/reference/ingestion-api#frequently_asked_questions).
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:216]:
```suggestion
				Add Gzip compression support to the `gcp_chronicle_unstructured` sink. See the [documentation](https://cloud.google.com/chronicle/docs/reference/ingestion-api#frequently_asked_questions).
```

File: website/cue/reference/releases/0.43.0.cue [hunk 11]
```diff
@@ -0,0 +1,449 @@
...
+				"""
+			contributors: ["chocpanda"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				The `sample` transform now has a `sample_rate_key` configuration option, which default to `sample_rate`, that allows configuring which key is used to attach the sample rate to sampled events. If set to an empty string, the sample rate will not be attached to sampled events.
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:223]:
```suggestion
				The `sample` transform now has a `sample_rate_key` configuration option, which default to `sample_rate`. It allows configuring which key is used to attach the sample rate to sampled events. If set to an empty string, the sample rate will not be attached to sampled events.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 12]
```diff
@@ -0,0 +1,449 @@
...
+				"""
+			contributors: ["dekelpilli"]
+		},
+		{
+			type: "chore"
+			description: """
+				The global `expire_metrics_secs` configuration option now defaults to 300s rather than being
```
Comment (neko-dd) [inline website/cue/reference/releases/0.43.0.cue:230]:
```suggestion
				The global `expire_metrics_secs` configuration option now defaults to `300s` rather than being
```

File: website/cue/reference/releases/0.43.0.cue (hunk 2 above)
Comment (pront) [reply website/cue/reference/releases/0.43.0.cue:34]: `Value` is an enum, `PartialCmp` tries to compare two `Value`s but those can actually hold a value of different type. We could actually shorten this to 

```
`vrl::value::Value` now implements `PartialCmp`
```

Comment (bfung) [reply website/cue/reference/releases/0.43.0.cue:34]:
```
... now implements `PartialOrd`
```
correct?

Comment (pront) [reply website/cue/reference/releases/0.43.0.cue:34]: This is the PR: https://github.com/vectordotdev/vrl/pull/1117/files. It basically added a trait implementation. On a second thought, this affect developers and not users. Maybe it is too low level for this changelog?

Comment (bfung) [reply website/cue/reference/releases/0.43.0.cue:34]: Yeah, and technically it's a VRL implementation change that also isn't used directly by end users.  Ok to exclude.

Comment (pront) [reply website/cue/reference/releases/0.43.0.cue]: Removed in https://github.com/vectordotdev/vector/pull/21906/commits/5ff6334d61e5a151fe1fa375bd2a8fb68f3de950.

File: website/cue/reference/releases/0.43.0.cue [hunk 13]
```diff
@@ -0,0 +1,447 @@
...
+			type: "feat"
+			description: """
+				Adds support for loading and concatenating multiple VRL files in the `remap` transform. This allows users to break down Vector remaps into smaller, more manageable units of configuration, improving organization, reusability, and maintainability of VRL code.
+				"""
+			contributors: ["brittonhayes"]
+		},
+		{
+			type: "fix"
+			description: """
+				Remove pubsub log only restriction. This allows using pubsub sink/source for metric and trace events.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:126]:
```suggestion
				The 'gcp_pubsub' sink now supports emitting metrics and traces.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 14]
```diff
@@ -0,0 +1,447 @@
...
+			type: "feat"
+			description: """
+				The `opentelemetry` source can now be configured to enrich log events with HTTP headers received in the OTLP/HTTP request.
+				"""
+			contributors: ["jblazquez"]
+		},
+		{
+			type: "feat"
+			description: """
+				Introduce a new `exclusive_route` transform, which functions as a switch statement to route events based on user-defined conditions.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:140]:
```suggestion
				Introduce a new `exclusive_route` transform, which functions as a switch statement to route events based on user-defined conditions. See the [release highlight](/highlights/2024-11-07-exclusive_route/) for more details on how to use this new transform.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 15]
```diff
@@ -0,0 +1,447 @@
...
+			type: "enhancement"
+			description: """
+				Pipeline name is now an optional configuration item for GreptimeDB log sink.
+				"""
+			contributors: ["sunng87"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				EDE code 30 (Invalid Query Type) (added in https://datatracker.ietf.org/doc/draft-ietf-dnsop-compact-denial-of-existence/04/) has been added to recognized EDE codes and has correct `purpose` attached to it.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:163]:
```suggestion
				The `dnstap` source now supports decoding of EDE code 30 (Invalid Query Type) (added in https://datatracker.ietf.org/doc/draft-ietf-dnsop-compact-denial-of-existence/04/) and has the correct `purpose` attached to it.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 16]
```diff
@@ -0,0 +1,447 @@
...
+				"""
+			contributors: ["esensar"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				Add VECTOR_HOSTNAME env variable to override the hostname used in the Vector events and internal metrics.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:170]:
```suggestion
				Add `VECTOR_HOSTNAME` env variable to override the hostname used in the Vector events and internal metrics.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 17]
```diff
@@ -0,0 +1,447 @@
...
+				This is useful when Vector is running on a system where the hostname is not meaningful, such as in a container (Kubernetes).
+				"""
+			contributors: ["nabokihms"]
+		},
+		{
+			type: "feat"
+			description: """
+				Add `update` support to the bulk API to Elasticsearch Sink.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:178]:
```suggestion
				The `elasticsearch` sink now supports publishing events as bulk "update"s by configuring `bulk.action` to `update`. This mode has a couple of constraints:
				
				- The message must be added in `.doc` and have `.doc_as_upsert` to true.
                                - `id_key` must be set, and the `encoding` field should specify `doc` and `doc_as_upsert` as values
```

File: website/cue/reference/releases/0.43.0.cue [hunk 18]
```diff
@@ -0,0 +1,447 @@
...
+				"""
+			contributors: ["blackrez"]
+		},
+		{
+			type: "feat"
+			description: """
+				Introducing the first version of the [OpenTelemetry](https://opentelemetry.io/docs/what-is-opentelemetry/) sink.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:185]:
```suggestion
				Introducing the first version of the [OpenTelemetry](https://opentelemetry.io/docs/what-is-opentelemetry/) sink. This initial implementation supports emitting logs as OTLP over HTTP. Support is expected to expand in the future.
```

File: website/cue/reference/releases/0.43.0.cue [hunk 19]
```diff
@@ -0,0 +1,447 @@
...
+			description: """
+				The `gelf` codec now correctly deserializes the subsecond portion of timestamps rather than dropping
+				them.
+				"""
+			contributors: ["jszwedko"]
+		},
+		{
+			type: "enhancement"
+			description: """
+				Support for watching config file changes by polling at certain interval.
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:251]:
```suggestion
				Support for watching config file changes by polling at certain interval rather than relying on notifications. This can be enabled setting `--watch-config-method` to `poll` where the interval can be configured via `--watch-config-poll-interval-seconds`.
```

**Comment by pront** [comment]
> Also, I think the base branch name here is wrong. It should be `v0.43`, not `v0.43.0`. The latter is what the git tag for the release will be.

👍 Fixed

**Comment by pront** [comment]
Minor but curious why this entry renders badly:
image

**Review by jszwedko — COMMENTED** [review]
Just a few more comments, otherwise looks good!

File: website/cue/reference/releases/0.43.0.cue [hunk 20]
```diff
@@ -184,7 +184,7 @@ releases: "0.43.0": {
 
 					1. The message must be added in `.doc` and have `.doc_as_upsert` to true.
 					2. `id_key` must be set, and the `encoding` field should specify `doc` and `doc_as_upsert` as values
-				"""
+					"""
```
Comment (jszwedko) [inline website/cue/reference/releases/0.43.0.cue:187]: I think you made the wrong indentation change here. The `"""`s were indented correctly. The text was indented one level too deep. See the entry above this one.

Comment (pront) [reply website/cue/reference/releases/0.43.0.cue]: I checked that it renders correctly both ways. Changed to https://github.com/vectordotdev/vector/pull/21906/commits/8e36c6eda15e2c7a6742e30010f2ab91a9c94479 for consistency.
